server/src/main/java/org/apache/cassandra/sidecar/tasks/PeriodicTaskExecutor.java

```diff
@@ -329,15 +343,18 @@ public String toString()
         }
     }
 
-    @VisibleForTesting
-    Map<PeriodicTaskKey, Long> timerIds()
+    enum Result
     {
-        return timerIds;
+        /**
+         * Tells the run completion handler whether task is rescheduled.
+         * If it is rescheduled, the delay is adjusted to the initial delay of the task
+         */
+        RESCHEDULED
```

Comment:
Can we have a Result that _doesn't_ mean RESCHEDULED, like COMPLETED? The fact that we  call`Promise#tryComplete` with either RESCHEDULED or nothing at all (essentially, `null`) seems really odd, even if we never really check for COMPLETED.

Reply:
`COMPLETED` is unnecessary. When the handler at `onComplete` is invoked, we know that it is completed.

Reply:
We can probably rename the enum. It is not really the result of execution. Looking for suggestions.

Reply:
Yeah - the biggest thing to me was that this enum is used as the type of an AsyncResult<T> and we set the result to `RESCHEDULED` or nothing at all, which was just jarring when I was looking at the code.

What do you think of actually using the `ScheduleDecision` enum here, and just getting rid of the `Result` all together? Then each branch of the `executeInternal` method would actually be able to indicate which decision was made - we end up only _using_ `RESCHEDULE` right now but at least there isn't an odd mix of `promise.tryComplete()` and `promise.tryComplete(Result.RESCHEDULE)` which didn't make a lot of sense to me - if the promise is a `Promise<Result>` it should really be completed with _something_ other than null... using `Promise<ScheduleDecision>` makes more sense.

Reply:
yep. `ScheduleDecision` makes better sense.